cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt

```diff
@@ -474,15 +464,11 @@ private class CampaignManagerImpl(
         }
         set(value) {
             val serialised = value?.let { JsonConverter.converter.encodeToString(value) }
-            dataStorage.ccpaConsentStatus = serialised
-            dataStorage.gppData = value?.gppData
-
-            // regenerate and update US privacy string with new values in the data storage
-            dataStorage.uspstring = generateCcpaUspString(
-                applies = value?.applies,
-                ccpaStatus = value?.status,
-                signedLspa = value?.signedLspa,
-            )
+            dataStorage.run {
+                ccpaConsentStatus = serialised
+                gppData = value?.gppData
+                uspstring = value?.uspstring
```

Comment:
I'm not sure if we need this one, since we don't want to count on the `uspString` from any backend response and we are going to generate `uspString` anyways, but maybe I'm missing something here

Reply:
We are going to update the `uspString` just once and only in one place, which is in the the method to handle the logic for meta data, this is just a simple assignment, it is just a set

Reply:
Not sure I understand the full context. But uspString needs to be updated in more places than only in the response of `/meta-data`. 

What happens if the user opens the CCPA pm and rejects all, for example?